tests/test_ad.py

```diff
@@ -62,25 +62,28 @@ def getRootDSE(self, request, reply):
 
 class AbstractLdapActiveDirectoryTestCase:
     @defer.inlineCallbacks
-    def setUp(self):
+    def make_ldap_server(self, api):
         self.ldap_server = yield ensureDeferred(
             create_ldap_server(_ActiveDirectoryLDAPServer)
         )
+
+        self.auth_provider = create_auth_provider(
+            self.ldap_server,
+            api,
+            config=self.getConfig(),
+        )
+
+    def make_mock(self, side_effects):
```

Comment:
I think the name of this function is not very specific on the outside, so I'd likely call it e.g.

```suggestion
    def make_module_api_mock(self, side_effects):
```

(similar for `side_effects`, but I'm tempted to suggest they be removed anyway ..

Reply:
actually, if you stop passing in the argument, I'd likely merge this back in to `setUp`... I think

Reply:
I've merged it back into `setUp`